File: coding_study_python/정렬_K번째수.py
def solution(array, commands):
    answer = []
    
    for command in commands:
        i,j,k = command
        answer.append(sorted(array[i-1:j])[k-1])
    
    return answer

if __name__ == "__main__":
    array = [1, 5, 2, 6, 3, 7, 4]
    commands = [[2, 5, 3], [4, 4, 1], [1, 7, 3]]
    # return = [5, 6, 3]
    
    solution(array, commands)

File: coding_study_kotlin/정렬_K번째수.kt
class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        var answer = intArrayOf()

        for(t in commands){
            val i = t[0]-1
            val j = t[1]-1
            val k = t[2]-1
            answer = answer.plus(array.slice(i..(j)).sorted()[k])
        }

        return answer
    }
}